website/server/libs/stripePayments.js

```
@@ -226,20 +227,27 @@ api.cancelSubscription = async function cancelSubscription (options, stripeInc)
   if (!customerId) throw new NotAuthorized(i18n.t('missingSubscription'));
 
   // @TODO: Handle error response
-  let customer = await stripeApi.customers.retrieve(customerId);
+  let customer = await stripeApi.customers.retrieve(customerId).catch(function errorCatch (err) {
+    return err;
```

why this catch? it basically swallows the error, without it it would have bubbled up and returned an api error. This way it's ignored

---

I'd like to log it somewhere, but do we have a good way of doing that at present? (console logging is bad for Lint)

By catching the error, though, if the customer ID doesn't exist on the payment processor end--such as when the subscription has been remotely canceled due to failed payments--we will go ahead and cancel the subscription on the Habitica side. That fixes users seeing a bare API error when they try to cancel a dead Stripe sub, and helps us prune unpaid-for subscriptions pending better Webhook handling. @TheHollidayInn may have further insight!

---

Canceing and skipping this error is fine. We just need to log in a library.

---

Winston is pretty good for example.

---

okay! We can use the logger in libs/logger to log it